left_effort_finger_trajectory_controller:
  type: effort_controllers/JointTrajectoryController
  constraints:
    goal_time: 1.0
    left_joint_finger_1: {goal: 0.02, trajectory: 0.05}
    left_joint_finger_2: {goal: 0.02, trajectory: 0.05}
    left_joint_finger_3: {goal: 0.02, trajectory: 0.05}
    stopped_velocity_tolerance: 0.02
  gains:
    left_joint_finger_1: {p: 1, d: 0.001, i: 0, i_clamp: 1}
    left_joint_finger_2: {p: 1, d: 0.001, i: 0, i_clamp: 1}
    left_joint_finger_3: {p: 1, d: 0.001, i: 0, i_clamp: 1}
  joints:
    - left_joint_finger_1
    - left_joint_finger_2
    - left_joint_finger_3
  stop_trajectory_duration: 0.5
  state_publish_rate: 25       
  action_monitor_rate: 10 


left_effort_finger_tip_trajectory_controller:
  type: effort_controllers/JointTrajectoryController
  constraints:
    goal_time: 1.0
    left_joint_finger_tip_1: {goal: 0.02, trajectory: 0.05}
    left_joint_finger_tip_2: {goal: 0.02, trajectory: 0.05}
    left_joint_finger_tip_3: {goal: 0.02, trajectory: 0.05}
    stopped_velocity_tolerance: 0.02
  gains:
    left_joint_finger_tip_1: {p: 0.5, d: 0, i: 0, i_clamp: 1}
    left_joint_finger_tip_2: {p: 0.5, d: 0, i: 0, i_clamp: 1}
    left_joint_finger_tip_3: {p: 0.5, d: 0, i: 0, i_clamp: 1}
  joints:
    - left_joint_finger_tip_1
    - left_joint_finger_tip_2
    - left_joint_finger_tip_3


left_position_joint_trajectory_controller:
  type: position_controllers/JointTrajectoryController
  constraints:
    goal_time: 0.6
    left_joint_1: {goal: 0.02, trajectory: 0.05}
    left_joint_2: {goal: 0.02, trajectory: 0.05}
    left_joint_3: {goal: 0.02, trajectory: 0.05}
    left_joint_4: {goal: 0.02, trajectory: 0.05}
    left_joint_5: {goal: 0.02, trajectory: 0.05}
    left_joint_6: {goal: 0.02, trajectory: 0.05}
    stopped_velocity_tolerance: 0.02
  gains:
    left_joint_1: {p: 5000, d: 0, i: 0, i_clamp: 10}
    left_joint_2: {p: 5000, d: 0, i: 0, i_clamp: 10}
    left_joint_3: {p: 5000, d: 0, i: 0, i_clamp: 10}
    left_joint_4: {p: 5000, d: 0, i: 0, i_clamp: 10}
    left_joint_5: {p: 5000, d: 0, i: 0, i_clamp: 10}
    left_joint_6: {p: 5000, d: 0, i: 0, i_clamp: 10}
  joints:
    - left_joint_1
    - left_joint_2
    - left_joint_3
    - left_joint_4
    - left_joint_5
    - left_joint_6
  stop_trajectory_duration: 0.5
  state_publish_rate: 25       
  action_monitor_rate: 10 

  #joint_state_controller:
  #  publish_rate: 50
  #  type: joint_state_controller/JointStateController

joint_state_controller:
  publish_rate: 100
  type: joint_state_controller/JointStateController

